pkgs/development/tools/build-managers/meson/default.nix

```diff
@@ -92,7 +95,24 @@ python3.pkgs.buildPythonApplication rec {
 
     # Do not propagate Python
     rm $out/nix-support/propagated-build-inputs
-  '';
+  '' + lib.optionalString (stdenv.hostPlatform != stdenv.targetPlatform) (
+    let
+      crossFile = writeText "cross-file.conf" ''
+        [binaries]
+        # fixes errors such as
+        # [Errno 8] Exec format error: './gdk3-scan'
+        exe_wrapper = ${lib.escapeShellArg (stdenv.targetPlatform.emulator buildPackages)}
```

Comment:
This maybe creates a big loop when you want to cross-compile a meson package and it's somehow a dependency of qemu. Then you need to rebuild hundreds of packages when you're just changing 1 file in a bad case :(. Not sure if we can easily prevent that, but that's really annoying if you're trying to fix something simple.

Reply:
It could be a separate hook that's added when needed

Reply:
I agree with Mindavi that if it is easily possible to move this to its own hook we should do that. I would imagine doing something similar like the CPython package does https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/cpython/default.nix#L367= 

I think we should also add it to passthru in case we might want to use the hook somewhere else.

Reply:
done